demo/deploy-vault.sh

```diff
@@ -27,9 +27,6 @@ spec:
       labels:
         app: vault
     spec:
-      nodeSelector:
-        kubernetes.io/arch: amd64
-        kubernetes.io/os: linux
```

Comment:
Since this isn't an image we build, let's specify `affinity` here so we can match exactly what the image supports, which `docker manifest inspect vault:1.4.0` shows should be:
- linux/amd64
- linux/arm
- linux/arm64
- linux/386

We do something similar here:
https://github.com/openservicemesh/osm/blob/3a7c924c9ebdedf9513220fc5d1c527b933f71b9/charts/osm/values.yaml#L230-L248